### PR TITLE
Added confirmation window when switching between tabs in Settings section

### LINKED ISSUE #2717: "Reset" Button Resets all the Fields in Settings?
This is more a question than it is an actual issue. 
If user enters something into varius fields in SSL Configuration, Server SSL Certificate and Device SSL Certificate and then presses "Reset" button, everything gets reset. I suggest that every view would have a different reset button, because user can do seomething in one field and want sto reset this, butthe side effect is that everything else gets reset. This is meant if user does this accidentally - a lot of data is lost. 

**Testflow:**
1. Login to Kura as admin/admin
2. Go to Settings, change some fields in the tabs mentioned above
3. Click "Reset" button and check all the previously modified fields - they are all reset to previous value

**Expected behavior**
Every tab should have its own "reset" button in case end-user makes a mistake and presses it. 

**Screenshots**
/

Target Environment (please complete the following information):
Board: Raspberry Pi 3
OS version: Linux raspberrypi 4.19.75-v7+ # 1270 SMP Tue Sep 24 18:45:11 BST 2019 armv7l GNU/Linux
Additional info: /

### PR DESCRIPTION
**Pull request by pierantoniomerlino**
This PR modifies the behavior of the Settings section, adding messages for alerting the user that he'll lose the filled data when switch to another tab.

**Related Issue:** This PR fixes/closes #2717

### CODE REVIEW

**Comment by MMaiero** [comment]
I believe Pierantonio and I are working on the same things: see #2972. Probably due to duplicate issues? Or I am wrong?

**Comment by pierantoniomerlino** [comment]
@MMaiero The issues were partially tracked in #2717 where we agreed on adding some warning messages for the user when he changes tab. So, yes, it's a (partial) duplicate issue...

**Comment by MMaiero** [comment]
I believe we can then close my PR and use this as it seems more complete

**Review by MMaiero — APPROVED** [review]

File: kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/client/ui/AbstractServicesUi.java [hunk 1]
```diff
@@ -86,9 +86,9 @@
 
     protected HashMap<String, Boolean> valid = new HashMap<>();
 
-    protected abstract void setDirty(boolean flag);
+    public abstract void setDirty(boolean flag);
```
Comment (LeoNerdoG) [inline kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/client/ui/AbstractServicesUi.java:89]: Please change header to "2020".

File: kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/client/ui/ConfigurableComponentUi.java [hunk 2]
```diff
@@ -72,7 +72,7 @@ public void setDirty(final boolean flag) {
     }
 
     @Override
-    protected void reset() {
+    public void reset() {
```
Comment (LeoNerdoG) [inline kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/client/ui/ConfigurableComponentUi.java:75]: Please change header to "2020".

File: kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/client/ui/drivers/assets/AssetConfigurationUi.java [hunk 3]
```diff
@@ -356,7 +356,7 @@ protected void renderTextField(final GwtConfigParameter param, boolean isFirstIn
     }
 
     @Override
-    protected void reset() {
+    public void reset() {
```
Comment (LeoNerdoG) [inline kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/client/ui/drivers/assets/AssetConfigurationUi.java:359]: Please change header to "2020".

File: kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/client/ui/settings/CommandUserTabUi.java [hunk 4]
```diff
@@ -18,6 +18,7 @@
 import org.eclipse.kura.web.client.messages.Messages;
 import org.eclipse.kura.web.client.ui.AbstractServicesUi;
 import org.eclipse.kura.web.client.ui.EntryClassUi;
+import org.eclipse.kura.web.client.ui.Tab;
```
Comment (LeoNerdoG) [inline kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/client/ui/settings/CommandUserTabUi.java:21]: Please change header to "2020".

File: kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/client/ui/settings/SettingsPanelUi.java [hunk 5]
```diff
@@ -16,17 +16,27 @@
 
 import org.eclipse.kura.web.client.messages.Messages;
 import org.eclipse.kura.web.client.ui.Tab;
+import org.eclipse.kura.web.client.ui.Tab.RefreshHandler;
```
Comment (LeoNerdoG) [inline kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/client/ui/settings/SettingsPanelUi.java:19]: Please change header to "2020".

File: kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/client/ui/settings/SslTabUi.java [hunk 6]
```diff
@@ -18,6 +18,7 @@
 import org.eclipse.kura.web.client.messages.Messages;
 import org.eclipse.kura.web.client.ui.AbstractServicesUi;
 import org.eclipse.kura.web.client.ui.EntryClassUi;
+import org.eclipse.kura.web.client.ui.Tab;
```
Comment (LeoNerdoG) [inline kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/client/ui/settings/SslTabUi.java:21]: Please change header to "2020".

**Review by LeoNerdoG — CHANGES_REQUESTED** [review]
Hey @pierantoniomerlino,
I have checked this PR. Code-wise please fix the headers. 
UI-wise: i found an issue (it happens on all of the tabs in Settings).

I think the easiest way will be through a Testflow:
1. Go to Settings tab, select one of "SSL" tabs
2. Enter something in on of the fields
3. Without applying changes click on another tab in Settings
4. When prompted to discarc changes select "No"
5. Again click on another tab and when prompted for discarding changes click "Yes"
6. Return to the previous tab and observe Apply and Reset buttons - they are both enabled! 

Can you please check on your side and confirm the issue?

**Comment by pierantoniomerlino** [comment]
@LeoNerdoG Yes, I confirm the issue

**Comment by pierantoniomerlino** [comment]
@LeoNerdoG now the issues should be fixed.

**Comment by LeoNerdoG** [comment]
Hey @pierantoniomerlino,
I have checked this and now it looks OK. I was on a wrong branch before. As far as I'm concerned, this can be merged.

**Pull request merged**